### PR TITLE
added call to aws to receive invoice data

### LINKED ISSUE #190: Get invoice data to precompile the update form
## Feature request
In the payment data update form the invoice data form must be pre-filled with the current fields previously entered by the user

### Feature description
To do this you need to call the aws backend which will retrieve the data from stripe/fattura24 for us

### How the feature is useful
The user will not need to remember the fields entered in a previous invoice

@simoneBrancati  write me the endpoint documentation when you are done

the data I need at the moment are all the invoice data : 

- invoice name

- invoice lastname

- invoice email 

- telephone

- country 

- city 

- address

- vat code

- fiscal code

- company name

- postal code

- province

- destination code

### PR DESCRIPTION
**Pull request by cpalessio96**
### All Submissions:

*   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
*   [x] Does your code has proper inline documentation? 

### Changes proposed in this Pull Request:
add call to aws to receive invoice data and precompile the invoice form

Closes #190  .

### CODE REVIEW

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- includes/Dashboard/PaymentTemplate.php  2
- src/stripe/Components/CheckoutForm.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5886550&bid=19269133)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- includes/Dashboard/PaymentTemplate.php  2
- src/stripe/Components/CheckoutForm.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5886550&bid=19269133)

File: includes/Utils/InvoiceHelper.php [hunk 1]
```diff
@@ -140,4 +140,56 @@ public function italian_invoice( $invoice_details, $invoice_object ) {
...
+			'fiscal_code'      => $response_body->data->invoiceDetails->FiscalCode,
+			'destination_code' => $response_body->data->invoiceDetails->DestinationCode,
+		);
+		$payment_data  = array(
+			'name'            => $response_body->data->paymentDetails->name,
+			'email'           => $response_body->data->paymentDetails->email,
+			'expiration'      => $response_body->data->paymentDetails->expiration,
+			'cardNumber'      => $response_body->data->paymentDetails->card,
+			'paymentMethodId' => $response_body->data->paymentDetails->paymentMethodId,
+		);
```
Comment (SalvoCozzubo) [inline includes/Utils/InvoiceHelper.php:189]: non c'è nessun controllo se `invoiceDetails` e `paymentDetails` esistono all'interno della risposta?
Per una maggiore leggibilità puoi usare una variabile di appoggio come `$invoiceDetails = $response_body->data->invoiceDetails` e quando vai a comporre l'oggetto richiami solo `$invoiceDetails->Name`

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- includes/Dashboard/PaymentTemplate.php  2
- src/stripe/Components/CheckoutForm.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5886550&bid=19269133)

File: includes/Utils/InvoiceHelper.php [hunk 2]
```diff
@@ -140,4 +140,61 @@ public function italian_invoice( $invoice_details, $invoice_object ) {
...
+		$code_response = wp_remote_retrieve_response_code( $response );
+		if ( 200 !== $code_response ) {
+			return false;
+		}
+		$response_body = json_decode( wp_remote_retrieve_body( $response ) );
+		if ( ! isset( $response_body->data->invoiceDetails, $response_body->data->paymentDetails ) ) {
+			return false;
+		}
+		$invoice_details = $response_body->data->invoiceDetails;
```
Comment (GiuseppeIuculano) [inline includes/Utils/InvoiceHelper.php:170]: Codacy found an issue: [Avoid unused local variables such as '$invoice_details'.](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5886550)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- includes/Dashboard/PaymentTemplate.php  2
- src/stripe/Components/CheckoutForm.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5886550&bid=19269133)

File: includes/Utils/InvoiceHelper.php [hunk 3]
```diff
@@ -140,4 +140,61 @@ public function italian_invoice( $invoice_details, $invoice_object ) {
...
+		$payment_details = $response_body->data->paymentDetails;
```
Comment (GiuseppeIuculano) [inline includes/Utils/InvoiceHelper.php:171]: Codacy found an issue: [Avoid unused local variables such as '$payment_details'.](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5886550)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/stripe/Components/CheckoutForm.js  2
- includes/Dashboard/PaymentTemplate.php  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=5886550&bid=19269133)

**Pull request merged**